pipeline/id_minter/src/main/scala/weco/pipeline/id_minter/database/IdentifiersDao.scala

```diff
@@ -67,16 +63,15 @@ class IdentifiersDao(identifiers: IdentifiersTable) extends Logging {
                       }
                     }
                 }.map(rs => {
-                    val row = getRowFromResult(i, rs)
-                    identifierRowsMap.get(row) match {
-                      case Some(sourceIdentifier) =>
-                        (sourceIdentifier, Identifier(i)(rs))
-                      case None =>
-                        // this should be impossible in practice
-                        throw new RuntimeException(
-                          s"The row returned by the query ($row) could not be matched to a sourceIdentifier in $sourceIdentifiers")
+                    val sourceIdentifier = getSourceIdentifierFromResult(i, rs)
+
+                    if (distinctIdentifiers.contains(sourceIdentifier)) {
+                      (sourceIdentifier, Identifier(i)(rs))
+                    } else {
+                      // this should be impossible in practice
```

Comment:
I love it when these go bang, one in the calm deletions checker went a while ago too